lnwallet/channel.go

```diff
@@ -4481,6 +4500,13 @@ func genHtlcSigValidationJobs(localCommitmentView *commitment,
 					"signatures")
 			}
 
+			// If this is a taproot channel, then we'll convert it
+			// to a schnorr signature, so we can get correct type
+			// from ToSignature below.
+			if chanType.IsTaproot() {
+				htlcSigs[i].ForceSchnorr()
```

Comment:
Yeah this allows us to use the same message and field, you just need to know contextually (based on the channel type) if this is a taproot channel or nto. 

> why would we do that instead of just creating new messages that use schnorr sigs directly?

Good question, we can certainly do that as well, but the initial decision was that the re-use of the field wasn't so bad. FWIW, the only diff with this new message would be that the `commit_sig` field is now that nonce+sig value we added earlier in the series.